pkg/kubelet/kubelet.go

```diff
@@ -273,11 +275,30 @@ func makePodSourceConfig(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps
 		glog.Infof("Adding manifest url %q with HTTP header %v", kubeCfg.ManifestURL, manifestURLHeader)
 		config.NewSourceURL(kubeCfg.ManifestURL, manifestURLHeader, nodeName, kubeCfg.HTTPCheckFrequency.Duration, cfg.Channel(kubetypes.HTTPSource))
 	}
+
+	// This is a one-shot fill the pipeline on startup if checkpoints exist
+	if kubeCfg.CheckpointPath != "" {
+		glog.Infof("Adding checkpoint path: %v", kubeCfg.CheckpointPath)
+		cmgr = checkpoint.NewCheckpointManager(kubeCfg.CheckpointPath)
+		c := cfg.Channel(kubetypes.ApiserverSource)
+		pods, err := cmgr.LoadPods()
+		if err != nil {
+			return nil, nil, err
+		}
+		if len(pods) > 0 {
+			c <- kubetypes.PodUpdate{Pods: pods, Op: kubetypes.SET, Source: kubetypes.ApiserverSource}
```

Comment:
are you sure? this will add the ApiserverSource to the sourcesReady list before we've actually gotten a full list from the API server. That allows AllReady() to return true, which allows housekeeping to run HandlePodCleanups() prior to us knowing what we should be running from the API server. Won't that delete running containers?